python_modules/libraries/dagster-msteams/dagster_msteams/resources.py

```diff
@@ -1,25 +1,76 @@
-from dagster import Bool, Field, Float, StringSource, resource
+from dagster import ConfigurableResource, resource
+from pydantic import Field
 
 from dagster_msteams.client import TeamsClient
 
 
-@resource(
-    {
-        "hook_url": Field(
-            StringSource,
-            description="""To send messages to MS Teams channel, an incoming webhook has to
-                    be created. The incoming webhook url must be given as a part of the
-                    resource config to the msteams_resource in dagster.
-                    """,
+class MSTeamsResource(ConfigurableResource):
+    """This resource is for connecting to Microsoft Teams.
+
+    The resource object is a `dagster_msteams.TeamsClient`.
+
+    By configuring this resource, you can post messages to MS Teams from any Dagster solid:
```

Comment:
```suggestion
    By configuring this resource, you can post messages to MS Teams from any Dagster asset/op:
```

Reply:
ha in https://github.com/dagster-io/dagster/pull/13464/files, i have it as `any Dagster op, asset, schedule or sensor.`

Reply:
^ that's better